net/service/handlers.go

```diff
@@ -1486,6 +1558,102 @@ func (service *OpenBazaarService) handleStore(pid peer.ID, pmes *pb.Message, opt
 	return m, nil
 }
 
+func (service *OpenBazaarService) handleOrderPayment(peer peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	// Unmarshall
+	if pmes.Payload == nil {
+		return nil, errors.New("payload is nil")
+	}
+	paymentDetails := new(pb.OrderPaymentTxn)
+	err := ptypes.UnmarshalAny(pmes.Payload, paymentDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	wal, err := service.node.Multiwallet.WalletForCurrencyCode(paymentDetails.GetCoin())
+	if err != nil {
+		return nil, err
+	}
+
+	wal0, ok := wal.(wallet.WalletMustManuallyAssociateTransactionToOrder)
+
+	if !ok {
+		return nil, nil
+	}
+
+	chash, err := chainhash.NewHashFromStr(paymentDetails.GetTransactionID())
+	if err != nil {
+		return nil, err
+	}
+
+	txn, err := wal.GetTransaction(*chash)
+	if err != nil {
+		return nil, err
+	}
+
+	contract, _, _, _, _, _, err := service.datastore.Sales().GetByOrderId(paymentDetails.OrderID)
+	if err != nil {
+		return nil, net.OutOfOrderMessage
+	}
+
+	if contract.VendorOrderConfirmation != nil &&
+		contract.BuyerOrder.Payment.Method != pb.Order_Payment_MODERATED {
+
+		// the seller has confirmed the order, so a simple check of
+		// the addresses and we are good to proceed
+		if !u.AreAddressesEqual(contract.VendorOrderConfirmation.PaymentAddress, txn.ToAddress) {
+			log.Debugf("mismatched payment address details: orderID: %s, expectedAddr: %s, actualAddr: %s",
+				paymentDetails.OrderID, contract.VendorOrderConfirmation.PaymentAddress, txn.ToAddress)
+			return nil, errors.New("mismatched payment addresses")
+
+		}
+
+	} else {
+		// the seller has not confirmed, so we need to compare the
+		// peerID in the vendorListing to the node peerID
+		if !(contract.VendorListings[0].VendorID.PeerID ==
+			service.node.IpfsNode.Identity.Pretty()) {
+			log.Debugf("mismatched peerID. wrong node is processing : orderID: %s, contractPeerID: %s",
+				paymentDetails.OrderID, contract.VendorListings[0].VendorID.PeerID)
+			return nil, errors.New("mismatched peer id")
+		}
+	}
+
+	toAddress, _ := wal.DecodeAddress(txn.ToAddress)
+	//fromAddress, _ := wal.DecodeAddress(txn.FromAddress)
+	output := wallet.TransactionOutput{
+		Address: toAddress,
+		Value:   txn.Value,
+		Index:   1,
+		OrderID: paymentDetails.OrderID,
+	}
+
+	input := wallet.TransactionInput{}
+
+	if paymentDetails.WithInput {
+		input = wallet.TransactionInput{
+			OutpointHash:  []byte(txn.Txid[:32]),
+			OutpointIndex: 1,
+			LinkedAddress: toAddress,
+			Value:         txn.Value,
+			OrderID:       paymentDetails.OrderID,
+		}
+	}
+
+	cb := wallet.TransactionCallback{
+		Txid:      txn.Txid,
+		Outputs:   []wallet.TransactionOutput{output},
+		Inputs:    []wallet.TransactionInput{input},
```

Comment:
Inputs would represents `spends` leaving the multisig. Since this tx is funds going into the multisig it should not hurt anything if the inputs are omitted.